autogen/agentchat/contrib/content_agent.py

```diff
@@ -0,0 +1,483 @@
+
+        # Define the prompt templates for the classifiers
+        self.content_classifier_prompt = lambda title, content: f"Title: `{title}`, Data: ```{content}`"
+        self.metadata_classifier_prompt = (
+            lambda content: f"We are parsing html metadata to extract useful data. Should we hold onto this item? {content}."
+        )
+
+        # Define the metadata classifier
+        self.metadata_classifier = AssistantAgent(
+            "Metadata Classifier",
+            system_message=self.metadata_classifier_system_msg,
+            llm_config=self.small_llm_config,
+            max_consecutive_auto_reply=0,
+        )
+        self.metadata_classifier.register_reply(self, self.classifier_to_collector_reply, 1)
+
+        # Define the html content classifier
+        self.content_classifier = AssistantAgent(
+            "Content Classifier",
+            system_message=self.content_classifier_system_msg,
+            llm_config=self.small_llm_config,
+            max_consecutive_auto_reply=0,
+        )
+        self.content_classifier.register_reply(self, self.classifier_to_collector_reply, 1)
+
+    # Main entry point
+    def collect_content(self, recipient, messages, sender, config):
+        content_type, content = "", ""
+        all_links = []
+        for message in messages:
+            if message.get("role") == "user":
+                links = re.findall(
+                    r"http[s]?://(?:[a-zA-Z]|[0-9]|[$-_@.&+]|[!*\\(\\),]|(?:%[0-9a-fA-F][0-9a-fA-F]))+",
+                    message.get("content"),
+                )
+                for link in links:
+                    all_links.append(link)
+
+        # Process the links provided by the user
+        for link in all_links:
+            content_type, content = self.fetch_content(link)
+
+        # Inform self that it has completed the root level of link(s)
+        self.link_depth = 1
+        if self.link_depth <= self.max_depth:
+            while len(self.additional_links) > 0:
+                additional_link = self.additional_links.pop()
+                content_type, content = self.fetch_content(additional_link)
+                all_links.append(all_links)
+
+        self.link_depth = 0
+        return (
+            True,
+            f"Success: archived the following links in your chosen location {self.local_dir}/ <-- {', '.join(all_links)}",
+        )
+
+    def fetch_content(self, link):
+        # Parse the link
+        parsed_url = urlparse(link)
+
+        # A special case for arxiv links
+        if "arxiv" in link and IS_ARXIV_CAPABLE:
+            return "pdf", self.fetch_arxiv_content(parsed_url)
+
+        elif parsed_url.path.endswith(".pdf"):
+            return "pdf", self.fetch_pdf_content(link)
+
+        else:
+            return "html", self.fetch_html_content(link)
+
+    def fetch_html_content(self, link):
```

Comment:
fetch_html_content -----> _fetch_html_content. Change the remaining private methods